### PR TITLE
add support for mesh using nftables by passing flag to agent config

### LINKED ISSUE #2272: Support for Istio Mesh + nft
When users use Istio Mesh + nftables (not iptables legacy) our rules run before mesh, leading to undefined behavior.
We probably should have a flag - iptables_mode which can be either legacy or nft then we use the appropriate and all should work the same (in theory).
Need to manually test it to know better though.
Linkerd has similar modes. `iptablesMode`

### PR DESCRIPTION
**Pull request by aviramha**
Closes #2272

### CODE REVIEW

File: mirrord/config/src/agent.rs [hunk 1]
```diff
@@ -253,6 +253,15 @@ pub struct AgentConfig {
     #[config(default = false)]
     pub privileged: bool,
 
+    /// ### agent.nftables {#agent-nftables}
+    ///
+    /// Use iptables-nft instead of iptables-legacy.
+    /// Defaults to `false`.
+    ///
+    /// Needed if your mesh uses nftables instead of iptables-legacy,
```
Comment (meowjesty) [inline mirrord/config/src/agent.rs:261]: Only for mesh?

File: mirrord/agent/src/steal/ip_tables.rs [hunk 2]
```diff
@@ -117,6 +117,17 @@ pub struct IPTablesWrapper {
     tables: Arc<iptables::IPTables>,
 }
 
+/// wrapper around iptables::new that uses nft or legacy based on env
+pub fn new_iptables() -> Result<IPTables, Box<dyn Error>> {
```
Comment (meowjesty) [inline mirrord/agent/src/steal/ip_tables.rs]: Do we have `Box<dyn Error>` anywhere else in the agent?

- If we do: then this gets a pass.
- If we don't: please wrap in an error.

**Review by meowjesty — CHANGES_REQUESTED** [review]

File: mirrord/config/src/agent.rs (hunk 1 above)
Comment (aviramha) [reply mirrord/config/src/agent.rs:261]: I don't know of other situations where you need (k8s doesn't use nftables for now)

File: mirrord/agent/src/steal/ip_tables.rs (hunk 2 above)
Comment (aviramha) [reply mirrord/agent/src/steal/ip_tables.rs]: It's just a propagation of the underlying error, didn't want to change the whole code for that.

File: mirrord/agent/src/steal/ip_tables.rs [hunk 3]
```diff
@@ -117,6 +117,17 @@ pub struct IPTablesWrapper {
...
+/// wrapper around iptables::new that uses nft or legacy based on env
+pub fn new_iptables() -> Result<iptables::IPTables, Box<dyn std::error::Error>> {
+    if let Some(val) = std::env::var("MIRRORD_AGENT_NFTABLES")
+        && val.lowercase() == "true"
+    {
+        iptables::new_with_cmd("/usr/sbin/iptables-nft")
+    } else {
+        iptables::new_with_cmd("/usr/sbin/iptables-legacy")
+    }
+}
```
Comment (meowjesty) [inline mirrord/agent/src/steal/ip_tables.rs:129]:
```suggestion
pub fn new_iptables() -> iptables::IPTables {
    if let Some(val) = std::env::var("MIRRORD_AGENT_NFTABLES")
        && val.lowercase() == "true"
    {
        iptables::new_with_cmd("/usr/sbin/iptables-nft")
    } else {
        iptables::new_with_cmd("/usr/sbin/iptables-legacy")
    }
    .expect("IPTables initialization may not fail!")
}
```

And remove the `unwrap`s from the other pieces of code. We're just `unwrap`ping this anyway right?

**Review by meowjesty — APPROVED** [review]
:+1:

**Pull request merged**